math/minuit2/src/MnContours.cxx

```diff
@@ -86,8 +86,66 @@ ContoursError MnContours::Contour(unsigned int px, unsigned int py, unsigned int
    // if Minos is not at limits we can use migrad to find the other corresponding point coordinate
    MnMigrad migrad0(fFCN, fMinimum.UserState(), MnStrategy(std::max(0, int(fStrategy.Strategy() - 1))));
 
+// function cross for a single parameter - copy the state
+   MnUserParameterState ustate = fMinimum.UserState();
+   MnFunctionCross cross1(fFCN, ustate, fMinimum.Fval(), fStrategy);
+   // need to copy the state since we modify it when calling function cross
+
+
+   auto findCrossValue = [&](unsigned int ipar, double startValue, double direction, bool & status) {
+      std::vector<unsigned int> vpar(1, ipar);
+      std::vector<double> vmid(1, startValue);
+      std::vector<double> vdir(1, direction);
+      ustate.Fix(ipar);
+      MnCross crossResult = cross1(vpar, vmid, vdir, toler, maxcalls);
+      status = crossResult.IsValid();
+      print.Debug("result of findCrossValue for par",ipar,"status: ",status," searching from ",vmid[0],"dir",vdir[0],
+                  " -> ",crossResult.Value()," fcn = ",crossResult.State().Fval());
+      return (status) ?  vmid[0] + crossResult.Value() * vdir[0] : 0;
+   };
+
+   // function to find teh contour points at the border of p1 after the minimization in p2
+   auto findContourPointsAtBorder = [&](unsigned int p1, unsigned int p2, double p1Limit, FunctionMinimum & minp2, bool order) {
+      // we are at the limit in p1
+      ustate.SetValue(p1, p1Limit);
+      ustate.Fix(p1);
+      bool ret1, ret2  = false;
+      double deltaFCN = fMinimum.Fval() + fFCN.Up()- minp2.UserState().Fval();
+      double pmid =  minp2.UserState().Value(p2);    // starting point for search
+      double pdir =  minp2.UserState().Error(p2) * deltaFCN/fFCN.Up(); // direction for the search
+      double y1 = findCrossValue(p2, pmid, pdir, ret1);
+      double y2 = findCrossValue(p2, pmid, -pdir, ret2);
+      if (!ret1 && !ret2) {
+         return std::vector<double>();
+      }
+      std::vector<double> yvalues; yvalues.reserve(2);
+      // check if value is
+      if (ret1 && !ret2) {
+          yvalues.push_back(y1);
+      } else if (!ret1 && ret2) {
+          yvalues.push_back(y2);
+      } else {
+         // otherwise use both points:
+         yvalues.push_back(y1);
+         // if points are not the same use both
+         if (std::abs(y1-y2) > 0.0001 * fMinimum.UserState().Error(p2)) {
+            // order them in increasing/decreasing y depending on order flag
+            int orderDir = (order) ? 1 : -1;
+            if (orderDir*(y2-y1) > 0)
+               yvalues.push_back(y2);
+            else {
+               yvalues.push_back(y2);
+               yvalues.push_back(y1);
```

Comment:
In this case, `yvalues` will contain `y1, y2, y1`. Is that the intent? The `print.Debug` statement below seems to suggest to only ever expect 2 values?

Reply:
Correct, thanks for finding this. It should be fixed in the next commit